zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkConnection.java

```diff
@@ -192,4 +234,59 @@ public String getServers() {
   public void addAuthInfo(String scheme, byte[] auth) {
     _zk.addAuthInfo(scheme, auth);
   }
+
+  private void lookupGetChildrenMethod() {
```

Comment:
If people are using the original patch, is this going to be problem?

Reply:
for this reflection way to determine the code used. there are three possibilities of server.
1/ normal server 2/ server with your patch 3/ server with original path.

The point I thinking is that if 3/ is used by some team, make sure it can work with our client same as 1/?

Reply:
Thanks, @kaisun2000 for the clarification!

I don’t think we have to do this. Firstly, we are solving the case for internal use cases. So we have our own ZK with pagination. Helix is by default still using apache ZK.
If the external users customize their ZK by directly using the original patch, I don’t think they would expect helix to be compatible with their custom ZK. So Helix will still call the non-paginated getChildren API that is supposed to be in their custom ZK. If they really want the pagination API in helix, it's recommended to use linkedin ZK or our patch, so helix will be compatible with that.